src/EmailAddressValidator.php

```diff
@@ -0,0 +1,295 @@
+
+                return false;
+            }
+
+            return $validHostname;
+        }
+
+        if (! $hostValidator->isValid($hostname)) {
+            $this->addMessage(sprintf(
+                $this->translate('%s is not a valid domain name for email address %s.'),
+                $hostname,
+                $email
+            ));
+
+            return false;
+        } elseif ($this->mx) {
+            // MX check on hostname
+            $hostname = $this->validateMXRecords($hostname, $email);
+        } else {
+            $hostname = idn_to_ascii($hostname);
+        }
+
+        return $hostname;
+    }
+
+    public function validateIp($value)
+    {
+        if (! filter_var($value, FILTER_VALIDATE_IP)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Returns true if and only if $value is a valid email address
+     * according to RFC2822
+     *
+     * @link   http://www.ietf.org/rfc/rfc2822.txt RFC2822
+     * @link   http://www.columbia.edu/kermit/ascii.html US-ASCII characters
+     * @param string $value
+     * @return boolean
+     */
+    public function isValid($value)
+    {
+        $this->clearMessages();
+
+        $matches = array();
+        $length = true;
+
+        // Split email address up and disallow '..'
+        if (
+            (strpos($value, '..') !== false)
+            || (! preg_match('/^(.+)@([^@]+)$/', $value, $matches))
+        ) {
+            $this->addMessage(sprintf(
+                $this->translate("'%s' is not a valid email address in the basic format local-part@hostname"),
+                $value
+            ));
+            return false;
+        }
+
+        $this->localPart = $matches[1];
+        $this->hostname = $matches[2];
+
+        $localPart = $matches[1];
+        $hostname = $matches[2];
+
+        if ((strlen($localPart) > 64) || (strlen($hostname) > 255)) {
+            $length = false;
+            $this->addMessage(sprintf(
+                $this->translate("'%s' exceeds the allowed length"),
+                $value
+            ));
+        }
+
+        // Match hostname part
+        $hostname = $this->validateHostnamePart($hostname, $value);
+
+        $local = $this->validateLocalPart($localPart, $value);
+
+        // If both parts valid, return true
+        if (($local && $hostname) && $length) {
+            return true;
+        }
+
+        return false;
+    }
+
+    protected function validateMXRecords($hostname, $email)
+    {
+        $mxHosts = [];
+        //decode IDN domain name
+        $decodedHostname = idn_to_ascii($hostname);
+
+        $result = getmxrr($decodedHostname, $mxHosts);
+        if (! $result) {
+            $this->addMessage(sprintf(
+                $this->translate("'%s' does not appear to have a valid MX record for the email address '%s'"),
+                $hostname,
+                $email
+            ));
+        } elseif ($this->deep && function_exists('checkdnsrr')) {
```

Comment:
Is there really still a chance this function does not exist? The PHP documentation does not mention that it belongs to an extension or anything :man_shrugging:

Reply:
True, this is not required and I have removed this check now.